packages/astro/src/runtime/server/render/component.ts

```diff
@@ -51,7 +51,7 @@ function isFragmentComponent(Component: unknown) {
 }
 
 function isHTMLComponent(Component: unknown) {
-	return Component && typeof Component === 'object' && (Component as any)['astro:html'];
+	return Component && (Component as any)['astro:html'];
```

Comment:
```suggestion
	return Component && (Component as any)['astro:html'] === true;
```

Changing to this fixes it for me. It's because of this proxy component that's tripping the check:

https://github.com/withastro/astro/blob/25c50300b52c680b64eea6d6a10557ec17029a73/packages/astro/test/fixtures/solid-component/src/components/ProxyComponent.jsx#L9-L14

where `Component['astro:html']` would return an actual JSX component.

Reply:
Ooh, that's hilarious. I saw the Proxy component but missed this, thank you very much!